src/AR-session.ts

```diff
@@ -7,51 +7,68 @@ import {ARProvider} from './AR-provider.js';
  * - handles global callbacks when AR session is started, ended
  * - can end any running AR session.
  */
-abstract class ARSession {
+
+class ARSession {
```

Comment:
We could still rename this to `AR` and separate this "static session manager and factor" from the ARSession itself, such that it's `AR.sessionForEngine(engine) => ARSession`

Reply:
That way I cannot prevent the user from manually creating ARSession. 
ARSession.sessionForEngine allows to call it's own private constructor.